Source/palette.cpp

```diff
@@ -271,6 +278,10 @@ void PaletteFadeOut(int fr)
  */
 static void CycleColors(int from, int to)
 {
+	extern int WaterDone;
+	if (WaterDone > 0)
+		return;

```

Comment:
Instead of modifying a general function to have quest specific exception, either modify palette_update_caves or even better make a copy of it specifically for this quest. Maybe it can even handle more of the quest specific events there.

Reply:
Excellent suggestion. This new version is indeed much cleaner.